extensions/amp-pinterest/0.1/pinit-button.js

```diff
@@ -0,0 +1,187 @@
+   * @param {String} str: the string to convert
+   * @returns {Element}
+   **/
+  stringToDOM(str) {
+    const temp = document.createElement('span');
+    temp/*REVIEW*/.innerHTML = str;
+    return temp.children[0];
+  }
+
+  /*
+   * Render helper for the optional count bubble
+   * @param {string} count: the data-count attribute
+   * @param {string} heightClass: the height class to apply for spacing
+   * @returns {string}
+   **/
+  renderCount(count, heightClass) {
+    Util.log('&type=pidget&button_count=1');
+    return `
+      <span class="-amp-pinterest-bubble-${this.count}${heightClass}">
+        ${this.formatPinCount(count)}
+      </span>
+    `;
+  }
+
+  /*
+   * Render the follow button
+   * @param {number} count: optional Pin count for the source URL
+   * @returns {Element}
+   **/
+  renderTemplate(count) {
+    const CLASS = {
+      shape: this.round ? '-round' : '-rect',
+      height: this.tall ? '-tall' : '',
+      lang: this.lang === 'ja' ? '-ja' : '-en',
+      color: ['red', 'white'].indexOf(this.color) !== -1 ? this.color : 'gray'
+    };
+
+    const clazz = [
+      `-amp-pinterest${CLASS.shape}${CLASS.height}`,
+      '-amp-fill-content'
+    ];
+
+    let countBubble = '';
+    if (!this.round) {
+      clazz.push(`-amp-pinterest${CLASS.lang}-${CLASS.color}${CLASS.height}`);
+      if (count) {
+        clazz.push(`-amp-pinterest-count-pad-${this.count}${CLASS.height}`);
+        countBubble = this.renderCount(count.count, CLASS.height);
+      }
+    }
+
+    return this.stringToDOM(`
+      <a class="${clazz.join(' ')}" href="${this.href}">
+        ${countBubble}
+      </a>
+    `);
+  }
+
+  /*
+   * Prepare the render data, create the node and add handlers
+   * @returns {Promise}
+   **/
+  render() {
+    this.color = this.element.getAttribute('data-color');
+    this.count = this.element.getAttribute('data-count');
+    this.lang = this.element.getAttribute('data-lang');
+    this.round = this.element.getAttribute('data-round');
+    this.tall = this.element.getAttribute('data-tall');
+
+    this.description = this.element.getAttribute('data-description');
+    this.description = encodeURIComponent(this.description);
+    this.media = encodeURIComponent(this.element.getAttribute('data-media'));
+    this.url = encodeURIComponent(this.element.getAttribute('data-url'));
+
+    const query = [
+      `amp=1`,
+      `guid=${Util.guid}`,
+      `url=${this.url}`,
+      `media=${this.media}`,
+      `description=${this.description}`
+    ].join('&');
+    this.href = `https://www.pinterest.com/pin/create/button/?${query}`;
+
+    let promise;
+    if (this.count === 'above' || this.count === 'beside') {
+      promise = this.fetchCount();
+    } else {
+      promise = Promise.resolve();
+    }
+    return promise
+      .then(this.renderTemplate.bind(this))
+      .then(pinitButton => {
+        pinitButton.onclick = this.handleClick.bind(this);
+        this.element.appendChild(pinitButton);
```

Comment:
I'd be happy to revisit later of more code continues to overlap, but I'd rather not introduce a whole new parent class for 1 or 2 lines of similarity.